google/cloud/alloydb/connector/instance.py

```diff
@@ -83,6 +82,25 @@ def __init__(
         self._current: asyncio.Task = self._schedule_refresh(0)
         self._next: asyncio.Task = self._current
 
+    def _parse_instance_uri(self, instance_uri: str) -> Tuple[str, str, str, str, str]:
+        # should take form "projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>"
+        if INSTANCE_URI_REGEX.fullmatch(instance_uri) is None:
+            raise ValueError(
+                "Arg `instance_uri` must have "
+                "format: projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>, "
+                f"got {instance_uri}."
+            )
+
+        instance_uri_split = INSTANCE_URI_REGEX.split(instance_uri)
+
+        return (
+            instance_uri,
+            instance_uri_split[1],
+            instance_uri_split[3],
+            instance_uri_split[5],
+            instance_uri_split[7],
```

Comment:
I don't think these are the proper indexes, are they? These indexes are assuming the old splitting of the instance URI based on the `/` in the string. The new split would be based on the regex matching pattern, same as [Go Connector](https://github.com/GoogleCloudPlatform/alloydb-go-connector/blob/ce45ab73f0d13ccd2a113196d13dc2ca326f52f4/internal/alloydb/instance.go#L81-L85). Adding the proper test cases will verify this. See the [Go test cases](https://github.com/GoogleCloudPlatform/alloydb-go-connector/blob/ce45ab73f0d13ccd2a113196d13dc2ca326f52f4/internal/alloydb/instance_test.go#L45) for help
```suggestion
            instance_uri_split[1],
            instance_uri_split[3],
            instance_uri_split[4],
            instance_uri_split[5],
```

Reply:
You are correct, the index was always out of index range. I have fixed it. Also I have added the test case for checking validity of the URI. I'll also add a invalid test to check if it fails correctly.